apps/extension/src/routes/popup/internal-msg-handler.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  isPopupRequest,
+  isTxApprovalReq,
+  PopupResponse,
+  TxApproval,
+} from '@penumbra-zone/types/src/internal-msg/popup';
+import { InternalMessageHandler } from '@penumbra-zone/types/src/internal-msg/shared';
+import { useStore } from '../../state';
+import { txApprovalSelector } from '../../state/tx-approval';
+
+export const popupMsgHandler = (
+  req: unknown,
+  _: chrome.runtime.MessageSender,
+  sendResponse: (x: unknown) => void,
+) => {
+  if (isPopupRequest(req)) {
```

Comment:
In the interest of not nesting `if`s and `try`s too deeply, I'd just return early here if this is false, e.g., 
```suggestion
  if (!isPopupRequest(req)) return;
```

Reply:
i constructed it this way to scope the `true` return, and make the fallback return a `false`, to improve the accuracy of intent to respond expressed to the runtime. we occasionally have subtle failures and this makes that failure space smaller.

Reply:
Yep, makes sense now — my comment here only applied if we were nixing the `true`/`false` return value